### PR TITLE
chore: improve pointers

### LINKED ISSUE #194: Improve pointers code
Some things that can be done:
- Simplify `Ok` functions
- Remove unnecessary `assembly` blocks

### PR DESCRIPTION
**Pull request by gnkz**
Closes #194

### CODE REVIEW

File: src/_modules/experimental/Request.sol [hunk 1]
```diff
@@ -356,17 +376,26 @@ function Ok(Request memory value) pure returns (RequestResult) {
     assembly {
         _value := value
     }
-    return RequestResult.wrap(Pointer.unwrap(ResultType.Ok.encode(_value)));
+    return ResultType.Ok.encode(_value).toRequestResult();
 }
 
 function Ok(Response memory value) pure returns (ResponseResult) {
     bytes32 _value;
```
Comment (vdrg) [inline src/_modules/experimental/Request.sol:383]: Maybe these could be `Pointer _value`? not really necessary but it makes sense for `encode` to receive a Pointer

**Review by vdrg — APPROVED** [review]
lgtm much cleaner!

File: src/_modules/experimental/Request.sol (hunk 1 above)
Comment (gnkz) [reply src/_modules/experimental/Request.sol]: Added some more simplifications here cd39646

**Review by vdrg — APPROVED** [review]
lgtm

**Pull request merged**